class/Services/RewriteService.php

```diff
@@ -29,8 +31,6 @@ public static function add_preview_rewrite()
 
     static::add_rewrite_rule($regex, $query);
     static::add_rewrite_tag('%passle_preview%');
-
-    flush_rewrite_rules();
```

Comment:
I think we actually do want this here. The issue with the old code is it was adding an action to the init event then flushing the rewrite rules, so it was being flushed before we'd added our new rule. Having it here is correct.